src/effects/VST3/VST3Effect.cpp

```diff
@@ -487,12 +387,8 @@ bool VST3Effect::RealtimeResume()
    return true;
 }
 
-bool VST3Effect::RealtimeProcessStart(EffectSettings &)
+bool VST3Effect::RealtimeProcessStart(EffectSettings& settings)
 {
-   assert(mPendingChanges == nullptr);
-   
-   //Same parameter changes are used among all of the realtime processors
-   mPendingChanges = mWrapper->mComponentHandler->getPendingChanges();
```

Comment:
Something is needed to replace the deleted lines, of course, so that there can be continuing adjustment of settings during play.  So it seems you would need more `ConsumeSettings` but in a worker thread.

I don't yet see the complete transformation of the code in later commits, but at this commit it is surely incomplete.

Reply:
Note the `RealtimeProcessStart` is the only place where the worker thread should check for changes of state communicated by the main thread.  Not in `RealtimeProcess` or `RealtimeProcessEnd`.

Reply:
Maybe the `EffectSettings` could contain a map from parameter ids to `optional<>` values, so the consuming thread could "clear" the map by reassigning `nullopt` everywhere, but not doing any free-store deallocation.

Reply:
>Something is needed to replace the deleted lines, of course, so that there can be continuing adjustment of settings during play.

Not neccessarily. VST3Effect doesn't yet (at this step) support realtime processing and doesn't support parameter changes "on the fly". So at this point it's enough to call ConsumeSettings once, just before processing has started.

>consuming thread could "clear" the map by reassigning nullopt everywhere, but not doing any free-store deallocation.

parameterChanges used to be a vector... but as I said in the previous comment, I'll move it to AssignSettings.